src/compile/mark/mark.ts

```diff
@@ -25,18 +26,39 @@ const markCompiler = {
   tick: tick,
   rule: rule,
   circle: circle,
-  square: square
+  square: square,
+  path: path
 };
 
 export function parseMark(model: UnitModel): any[] {
+  // TODO: new branch here parsePathMark()
+  let marks = null;
   if (contains([LINE, AREA], model.mark())) {
-    return parsePathMark(model);
+    marks = parseLineOrArea(model);
   } else {
-    return parseNonPathMark(model);
+    marks = parseNonLineOrArea(model);
```

Comment:
Sorry, I don't know what you're trying to ask.  Basically, I'm not sure what's the real different between the two.   
- The only think you change is `[marks]` and `marks` ?  Unless I misunderstood ...
- It might be more helpful to make atomic commit and point me to a specific commit and ask.